x-pack/plugins/uptime/server/rest_api/synthetics_service/edit_monitor.ts

```diff
@@ -40,11 +44,20 @@ export const editSyntheticsMonitorRoute: UMRestApiRouteFactory = () => ({
     const { syntheticsService } = server;
 
     try {
+      const previousMonitor: SavedObject<MonitorFields> = await savedObjectsClient.get(
+        syntheticsMonitorType,
+        monitorId
+      );
+      const monitorWithRevision = {
+        ...monitor,
+        revision: (previousMonitor.attributes[ConfigKey.REVISION] || 0) + 1,
+      };
```

Comment:
not sure why it needs to fetch the saved object?

Reply:
Because we need to know the previous revision number, in order to increment it.